lua/rest-nvim/utils/init.lua

```diff
@@ -48,27 +48,65 @@ end
 -- variables
 M.read_env_file = function()
   local variables = {}
-  local env_file = "/" .. (config.get("env_file") or ".env")
-
-  -- Directories to search for env files
-  local env_file_paths = {
-    -- current working directory
-    vim.fn.getcwd() .. env_file,
-    -- directory of the currently opened file
-    vim.fn.expand("%:p:h") .. env_file,
-  }
-
-  -- If there's an env file in the current working dir
-  for _, env_file_path in ipairs(env_file_paths) do
-    if M.file_exists(env_file_path) then
-      for line in io.lines(env_file_path) do
-        local vars = M.split(line, "=", 1)
-        variables[vars[1]] = vars[2]
-      end
-    end
-  end
 
-  return variables
+	-- If there is a line at the beginning with @ first
+	if vim.fn.search('^@', 'cn') > 0 then
+		-- Read all lines of the file
+		local lines = vim.api.nvim_buf_get_lines(0, 0, -1, true);
+
+		-- For each line
+		for _, line in pairs(lines) do
+			-- Get all that starts with @
+			if line:match("^@") then
+				-- Read them off
+				local place = M.split(line, '=', 1);
+				-- Remove whitespace around them
+				for i, each in ipairs(place) do
+					place[i] = each:gsub(" ", "");
+				end
+				-- Add to variables, and remove the @
+				variables[place[1]:gsub('@', '')] = place[2];
```

Comment:
There should be a `nil` check here otherwise it will throw errors.

Reply:
From what I can tell, there does not seem to be any need for a nil-check. There will always be 2 values in 'place'.

If the left side is not present, then it is not evaluated, if the right side is not present then a "" is placed there. string:gsub as well only removes if it exists.

Reply:
yeah, this comment was made before there was `^@` regex in place.